dired-narrow.el

```diff
@@ -95,6 +95,27 @@ we should act on."
                  (function :tag "Use custom function."))
   :group 'dired-narrow)
 
+(defcustom dired-narrow-exit-when-1-left nil
+  "If there is only one file left while narrowing,
+exit minibuffer and call dired-narrow-exit-action."
+  :group 'dired-narrow)
```

Comment:
`defcustom` also supports a `:type` argument.  Documentation on available types is here https://www.gnu.org/software/emacs/manual/html_node/elisp/Customization-Types.html#Customization-Types

Specifying a type will select proper widget for the user when they are customizing the setting.